week-1 tasks of book and user service

sskadam45 opened the pull request.

sskadam45 committed:
week-1 tasks of book and user service

rupeshpatil761 commented on book-service/pom.xml:
```diff
@@ -0,0 +1,65 @@
+		</plugins>
+	</build>
+
+</project>
```
Do we really need all the above files and changes for maven?
I think pom.xml is enough in each service.

rupeshpatil761 commented on book-service/src/main/java/com/jpop/bookservice/controller/BookController.java:
```diff
@@ -0,0 +1,52 @@
+        return bookService.findAll();
+    }
+
+    @GetMapping("books/{id}")
```
Try to follow versioning.
https://dzone.com/articles/versioning-rest-api-with-spring-boot-and-swagger

sskadam45 replied:
Resolved

rupeshpatil761 commented on user-service/src/test/java/com/jpop/userservice/UserServiceApplicationTests.java:
```diff
@@ -0,0 +1,13 @@
+	void contextLoads() {
+	}
+
+}
```
Please refer to book service review comments, same goes for user-service

rupeshpatil761 commented on book-service/src/main/java/com/jpop/bookservice/service/BookService.java:
```diff
@@ -0,0 +1,31 @@
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+}
```
You may handle exceptions in a common place.
May be Generic Exception Handler class

sskadam45 committed:
gitignore chnage

sskadam45 committed:
added git ignore file changes

sskadam45 committed:
gitignore chnage

sskadam45 committed:
gitignore changes

sskadam45 committed:
gitignore changes

sskadam45 committed:
gitignore changes

sskadam45 committed:
Book service:

-resolved Review comments
-Added change for other servicess like eureka,rabittmq,configserver etc

sskadam45 committed:
User-service:

-Resolved review comments
-Added changes for other services like eureka,rabitmq,config server etc

sskadam45 commented:
Resolved the review comments. Let me know if anything missing.

rupeshpatil761 commented:
Looks good to me
